### PR TITLE
Fix find logic

### LINKED ISSUE #59: Find command matching all keywords
FInd command matches **all** keywords instead of one

### PR DESCRIPTION
**Pull request by raageth**
resolves #59

### CODE REVIEW

**Comment by yockcheng** [comment]
LGTM!

**Review by yockcheng — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 1]
```diff
@@ -33,6 +38,25 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
...
+    }
+
+    /**
+     * Returns true if the {@code sentence} contains the {@code word}.
+     *   Ignores case, but a full word match is required.
+     *   <br>examples:<pre>
+     *       containsWordIgnoreCase("ABc def", "abc") == true
+     *       containsWordIgnoreCase("ABc def", "DEF") == true
+     *       containsWordIgnoreCase("ABc def", "AB") == false //not a full word match
+     *       </pre>
```
Comment (Domokunx) [inline src/main/java/seedu/address/commons/util/StringUtil.java:52]: Should there be empty line between description and param?

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 2]
```diff
@@ -69,4 +135,10 @@ public static boolean isNonZeroUnsignedInteger(String s) {
             return false;
         }
     }
+
```
Comment (Domokunx) [inline src/main/java/seedu/address/commons/util/StringUtil.java:138]: Is the main method for testing purposes? If so should be fine to remove

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 3]
```diff
@@ -126,6 +126,67 @@ public void containsWordIgnoreCase_validInputs_correctResult() {
...
+     *   - matches first word in sentence
+     *   - last word in sentence
+     *   - middle word in sentence
+     *   - matches multiple words
+     *   - sentence contains all characters in word (in order)
+     *
+     * Possible scenarios returning false:
+     *   - sentence word matches part of the query word
+     *
+     * The test method below tries to verify all above with a reasonably low number of test cases.
```
Comment (Domokunx) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:151]: Nice header!

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 4]
```diff
@@ -126,6 +126,67 @@ public void containsWordIgnoreCase_validInputs_correctResult() {
...
+     */
+
```
Comment (Domokunx) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:153]: I think no empty line between header and methods?

**Review by Domokunx — COMMENTED** [review]
LGTM!

Minor cosmetic changes to adhere to the style guide

File: src/test/java/seedu/address/commons/util/StringUtilTest.java (hunk 4 above)
Comment (raageth) [reply src/test/java/seedu/address/commons/util/StringUtilTest.java:153]: Will change but this was previous code from AB3.

File: src/main/java/seedu/address/commons/util/StringUtil.java (hunk 1 above)
Comment (raageth) [reply src/main/java/seedu/address/commons/util/StringUtil.java:52]: Will change but this was previous code from AB3.

**Pull request merged**